libafl/src/events/llmp/hooks/centralized.rs

```diff
@@ -0,0 +1,117 @@
+use std::{fmt::Debug, marker::PhantomData};
+
+#[cfg(feature = "llmp_compression")]
+use libafl_bolts::{compress::GzipCompressor, llmp::LLMP_FLAG_COMPRESSED};
+use libafl_bolts::{
+    llmp::{Flags, LlmpHook, LlmpMsgHookResult, Tag},
+    shmem::ShMemProvider,
+    ClientId, Error,
+};
+
+#[cfg(feature = "llmp_compression")]
+use crate::events::COMPRESS_THRESHOLD;
+use crate::{
+    events::{BrokerEventResult, Event, _LLMP_TAG_TO_MAIN},
+    inputs::Input,
+};
+
+/// An LLMP-backed event manager for scalable multi-processed fuzzing
+pub struct CentralizedLlmpHook<I, SP>
+where
+    I: Input,
+    SP: ShMemProvider + 'static,
+{
+    #[cfg(feature = "llmp_compression")]
+    compressor: GzipCompressor,
+    phantom: PhantomData<(I, SP)>,
+}
+
+impl<I, SP> LlmpHook for CentralizedLlmpHook<I, SP>
+where
+    I: Input,
+    SP: ShMemProvider + 'static,
```

Comment:
and this `SP` is not needed too.
but at this point you can just do fn on_new_message<I>